ui/src/shared/utils/filterUnusedVars.ts

```diff
@@ -50,10 +50,13 @@ export const createdUsedVarsCache = (variables: Variable[]) => {
 
 export const filterUnusedVarsBasedOnQuery = (
   variables: Variable[],
-  queryTexts: string[]
+  queryTexts: string[],
+  keepSystemVariables?: boolean
 ) => {
-  const varsInUse = variables.filter(variable =>
-    queryTexts.some(text => isInQuery(text, variable))
+  const varsInUse = variables.filter(
+    variable =>
+      queryTexts.some(text => isInQuery(text, variable)) ||
```

Comment:
Is it worth considering add some tests to `filterUnusedVars` below:

https://github.com/influxdata/influxdb/pull/19074/files#diff-8e45b03aeca264900ab0daa6f18aacd0R71

Since it's also dependent on this and might need to have an additional parameter passed in?